### PR TITLE
Allow non-negative replicas in 'kubernetes_stateful_set'.

### PR DESCRIPTION
**Pull request by bmcstdio**
This is my attempt at fixing #524.

### CODE REVIEW

**Comment by pdecat** [comment]
Do updates work scaling from > 0 to 0 given the following check?
https://github.com/terraform-providers/terraform-provider-kubernetes/blob/22f458f2e9d8965681ef8fac43d256c9d5900462/kubernetes/structures_stateful_set.go#L27

It could be worth validating with an additional acceptance test case.

Note: the same validation may also be used for:
- Deployment:
  - `replicas`: https://github.com/terraform-providers/terraform-provider-kubernetes/blob/9c3d1a4942a0c0a8a1285446a0b016260ee0dd92/kubernetes/resource_kubernetes_deployment.go#L68
  - `revision_history_limit`: https://github.com/terraform-providers/terraform-provider-kubernetes/blob/9c3d1a4942a0c0a8a1285446a0b016260ee0dd92/kubernetes/resource_kubernetes_deployment.go#L74
- ReplicationControllers:
  - `replicas`: https://github.com/terraform-providers/terraform-provider-kubernetes/blob/9c3d1a4942a0c0a8a1285446a0b016260ee0dd92/kubernetes/resource_kubernetes_deployment.go#L68
- StatefulSet:
  - `revision_history_limit`: https://github.com/terraform-providers/terraform-provider-kubernetes/blob/fbf2bd5cb8c94022ae2ef07b1ec720e081ce9e1e/kubernetes/schema_stateful_set_spec.go#L28

**Comment by bmcstdio** [comment]
> Do updates work scaling from > 0 to 0 given the following check?

Nice catch, I've amended that and added the requested acceptance test.

> Note: the same validation may also be used for:

That's correct, but maybe we can make that part of a separate PR?

File: kubernetes/resource_kubernetes_stateful_set_test.go [hunk 1]
```diff
@@ -148,6 +148,35 @@ func TestAccKubernetesStatefulSet_update_replicas(t *testing.T) {
...
+		Steps: []resource.TestStep{
+			{
+				Config: testAccKubernetesStatefulSetConfigBasic(name),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKubernetesStatefulSetExists(statefulSetTestResourceName, &conf),
+					testAccKubernetesStatefulSetChecksBasic(name),
+				),
+			},
+			{
+				Config: testAccKubernetesStatefulSetConfigUpdateReplicas(name),
```
Comment (pdecat) [inline kubernetes/resource_kubernetes_stateful_set_test.go]: Doesn't the config returned by that function set replicas to 5?

https://github.com/terraform-providers/terraform-provider-kubernetes/blob/9c3d1a4942a0c0a8a1285446a0b016260ee0dd92/kubernetes/resource_kubernetes_stateful_set_test.go#L645

Comment (bmcstdio) [reply kubernetes/resource_kubernetes_stateful_set_test.go]: It does, thank you - I have amended the commit.

**Comment by pdecat** [comment]
To quickly test your changes locally with minikube:

```
# export KUBE_CTX=minikube
# make testacc TESTARGS='-run=TestAccKubernetesStatefulSet_update_replicas_.*'
```

**Comment by bmcstdio** [comment]
Thanks @pdecat!

```
$ make testacc TESTARGS='-run=TestAccKubernetesStatefulSet_update_replicas_.*'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test "./kubernetes" -v -run=TestAccKubernetesStatefulSet_update_replicas_.* -timeout 120m
=== RUN   TestAccKubernetesStatefulSet_update_replicas_to_five
--- PASS: TestAccKubernetesStatefulSet_update_replicas_to_five (7.05s)
=== RUN   TestAccKubernetesStatefulSet_update_replicas_to_zero
--- PASS: TestAccKubernetesStatefulSet_update_replicas_to_zero (6.99s)
PASS
ok      github.com/terraform-providers/terraform-provider-kubernetes/kubernetes 14.092s
```

**Pull request merged**